### PR TITLE
Clear /etc/resolv.conf before building image

### PR DESCRIPTION
**Pull request by devpatha**
#### Why I did it
nameserver and domain entries from build system fsroot gets into sonic image.

#### How I did it
Clear /etc/resolv.conf before building image

#### How to verify it
Built image with it and verified with install that /etc/resolv.conf is empty

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205

### CODE REVIEW

File: build_debian.sh [hunk 1]
```diff
@@ -678,6 +678,7 @@ sudo rm -f $ONIE_INSTALLER_PAYLOAD $FILESYSTEM_SQUASHFS
 ## Note: -x to skip directories on different file systems, such as /proc
 sudo du -hsx $FILESYSTEM_ROOT
 sudo mkdir -p $FILESYSTEM_ROOT/var/lib/docker
+sudo LANG=C chroot $FILESYSTEM_ROOT /bin/bash -c "truncate -s 0 /etc/resolv.conf"
```
Comment (Blueve) [inline build_debian.sh:681]: Some users might intend to add some entries to `/etc/resolv.conf` for final build image.
This truncate might remove them unexpected. Maybe have a resolv.conf in `/files/image_config/resolv.conf` would be a better solution? We can override the file by using `/files/image_config/resolv.conf` instead truncate the file. And user can modify the image_config if they intent to have some entries in final image.

**Review by yxieca — APPROVED** [review]

**Review by Blueve — APPROVED** [review]

**Review by saiarcot895 — APPROVED** [review]

**Comment by yxieca** [comment]
Why I did it
nameserver and domain entries from build system fsroot gets into sonic image.

How I did it
Clear /etc/resolv.conf before building image

How to verify it
Built image with it and verified with install that /etc/resolv.conf is empty

**Comment by qiluo-msft** [comment]
This commit could not be cleanly cherry-picked to 202012. Please submit another PR.

**Comment by devpatha** [comment]
> This commit could not be cleanly cherry-picked to 202012. Please submit another PR.

I created https://github.com/sonic-net/sonic-buildimage/pull/12686 for 202012.

**Comment by kevinskwang** [comment]
> > This commit could not be cleanly cherry-picked to 202012. Please submit another PR.
> 
> I created #12686 for 202012.

Merged

**Pull request merged**